price-estimator/src/models.rs

```diff
@@ -130,6 +130,11 @@ impl From<&pricegraph::TransitiveOrderbook> for MarketsResult {
 #[serde(transparent)]
 pub struct PriceEstimateResult(pub Option<f64>);
 
+#[derive(Debug, Serialize)]
+pub struct ErrorMessage {
+    pub message: &'static str,
+}
```

Comment:
nit: I think the struct shouldn't be an `ErrorMessage` but more of an `Error`. Also I think the `message` field should be used for a human readable messages that may include some extra context and not status constants (which should probably be an enum instead of `&'static str`).
```suggestion
#[derive(Debug, Serialize)]
pub struct Error {
    pub status: ErrorStatus,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum ErrorState {
    NotFound,
    NoTokenInfo,
    // ...
}
```

Reply:
I disagree. This struct represents the body of the response we send when we do not respond with success. The `handle_rejection` function decides what goes in there so I see no reason to introduce an enum with more complicated serialization.
I agree with the human readable part so i'll change the messages a bit to contain more informatin.

Reply:
My main argument was that if the error message was going to be `NOT_FOUND` style string constants, then it should be an enum (and a `status` or `kind` as opposed to a `message`). This could be very valuable for clients to recover an exact error type from the error data (kind of like the `ErrorKind` pattern in Rust).

`messages` are intended to be human readable error messages, kind of like `Error::description` or `Error::to_string` in Rust.

Reply:
I'm not sure if we should make any guarantees about the contents of the error. Human readable has practical benefits but I don't think any client can recover in a meaningful way so I don't want to make more work for us to implement this. This is something I'm not super confident about so I could be convinced but I would also like to have that discussion in another issue/pr and keep this simple way of handling errors for now.

Reply:
> I would also like to have that discussion in another issue/pr

:+1: 

I just have a nit with the naming, maybe
```suggestion
#[derive(Debug, Serialize)]
pub struct ErrorResult {
    pub message: &'static str,
}
```

Reply:
One reason error kinds are nice is that they frontend can easily localize them to other languages (which a backend API usually doesn't). But I agree that it is not needed in our current state.